docs/DeveloperGuide.md

```diff
@@ -1,38 +1,462 @@
+`loadFiles(hasDirectory, hasPropertyFile, hasClientFile, hasPairingFile, clientList, propertyList, pairingList)`.
+This method would conduct the following operations:
+- Create a `data` directory if not already exist. (`hasDirectory` is `false`)
+- Load Clients into Client ArrayList if `hasClientFile` is `true`.
+- Load Properties into Property ArrayList if `hasPropertyFile` is `true`.
+- Load Pairings into Pairing HashMap if `hasPairingFile` is `true`.
+
+An empty file would not be loaded into the ArrayList and PropertyList as the code is designed to read for `next()`.
+An empty file would invoke a `false` in `hasNext()`, thus adding operation would not continue. The overall operation can
+be visualised in the flowchart above.
+
+#### Append To File
+When file is appended into the text file, it's being stored in different formats as shown below:
+
+- Client:  `NAME | CONTACT_NUMBER | EMAIL <optional> | BUDGET` 
+- Property: `NAME | ADDRESS | RENTAL_PRICE | UNIT_TYPE` 
+- Pairing: `[CLIENT_FORMAT] : [PROPERTY_FORMAT]` 
+
+The text file of which Client, Property and Pairing is being stored is `client.txt`, `property.txt` and `pairing.txt` 
+respectively.
+
+![Add Client to Storage Diagram](diagrams/StorageAddClientSequenceDiagram.jpg)
+![Add Property to Storage Diagram](diagrams/StorageAddPropertySequenceDiagram.jpg)
+![Add Pair to Storage Diagram](diagrams/StorageAddPairSequenceDiagram.jpg)
+
+The three sequence diagram above shows the sequence of which the append operation is being invoked. All three
+operations are similar in operations but are invoked with different `parameter` and `path`.
+
+### Update To File
+The update operation happens when entries in ClientList and PropertyList is being deleted and entries the hash map of 
+PairingList is being removed.
+
+The sequence diagram of `updateClient`, `updateProperty` and `updatePair` can be seen below:
+![Update Client Sequence Diagram](diagrams/StorageUpdateClientSD.jpg)
+![Update Property Sequence Diagram](diagrams/StorageUpdatePropertySD.jpg)
+![Update Pairing Sequence Diagram](diagrams/StorageUpdatePairSD.jpg)
+
+Note that when delete operation is being invoked on client and property, the `updatePair` method will also be invoked to
+prevent entries retaining within pairingList after it has been deleted from clientList or propertyList.
+
+
+### List feature
+
+There are 3 main steps whenever a list command needs to be executed
+* When the user enters any command, it first needs be understood. That is handled by the ParseManager class.
+   Next, when the first word entered by the user is determined to be `list`, ParseManager itself then determines
+   the type of list command entered, including the tags. ParseListClient, ParseListProperty and ParseListEverything
+   are checkers to ensure that a valid command has been entered. ParseListClient and ParseListProperty also determine
+   if tags have been entered, and if those tags are valid.
+  ```
+        case COMMAND_LIST:
```

Comment:
This chunk of code is confusing and does not seem to help with explanation. Furthermore, some aspects of the code is also not explained.